scripts/generate_ledgers_tar_from_config.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -e
+
+RUNTIME_GENESIS_LEDGER_EXE=${RUNTIME_GENESIS_LEDGER_EXE:-"./_build/default/src/app/runtime_genesis_ledger/runtime_genesis_ledger.exe"}
+
+if [ $# -lt 3 ]; then
+    echo "Usage: $0 <config-file-input> <genesis-dir> <config-file-output>"
+    exit 1
+fi
+
+config_file_input=$1
+genesis_dir=$2
+config_file_output=$3
+
+# preprocess config file
+config_file_tmp=$(mktemp)
+echo "proprocessing config file..."
+jq -r 'if .proof.fork | has("previous_global_slot") then .proof.fork.genesis_slot = .proof.fork.previous_global_slot | del(.proof.fork.previous_global_slot) else . end' "$config_file_input" >"$config_file_tmp"
+
+echo "generating genesis ledger... (this may take a while)"
+tmp_log_file=$(mktemp)
+grep_strings=("ledger_file" "tar_path" "root_hash")
+pattern=$(
+    IFS="|"
+    echo "${grep_strings[*]}"
+)
+$RUNTIME_GENESIS_LEDGER_EXE --config-file "$config_file_tmp" --genesis-dir "$genesis_dir" | tee >(grep -E "$pattern" >"$tmp_log_file")
+tmp_file=$(mktemp)
+echo "genesis ledger generated"
+
+function get_ledger_root_hash() {
+    ledger_name=$1
+    json=$(grep "$ledger_name" <"$tmp_log_file")
```

Comment:
Grepping logs this way is a bit fragile.

Could we make runtime genesis ledger tool accept an optional filename argument and write a json to that file with all the needed values?

Reply:
Yes, we can! I've introduced an optional `--hash-output-file` filename argument and the runtime genesis ledger tool writes the ledger hashes to there if the flag is present. The script retrieves the hashes from that file instead of parsing the logs.